xcode/Subconscious/Shared/Library/Func.swift

```diff
@@ -58,6 +62,8 @@ struct Func {
     ) async throws -> T? {
         do {
             return try await perform(attempts)
+        } catch RetryError.cancelled {
+            return nil
```

Comment:
Open question, but would it be better to have this retry with backoff logic be part of an update function instead? (If fx fails, send another after a growing delay).

The advantage would be that retry would be forced to fold with app state at each step. The actions would be sent through the logs, and current backoff value would be written to the model where we can inspect it.

Non-blocking.

Reply:
I can see the value of this but it could be a little over-complicated right now. 

The main issue I see is that multiple of these processes can be running at once (rapidly following multiple users before the last one resolves), so tracking the state will require attaching an ID to each background job etc.

I'd like to think a bit more about the right abstraction to make it trivial for different components to sign up for this functionality and share the code, put some initial thoughts down here: https://github.com/subconsciousnetwork/subconscious/issues/658